x-pack/plugins/enterprise_search/kibana.json

```diff
@@ -2,7 +2,7 @@
   "id": "enterpriseSearch",
   "version": "kibana",
   "kibanaVersion": "kibana",
-  "requiredPlugins": ["features", "licensing"],
+  "requiredPlugins": ["features", "licensing", "charts"],
```

Comment:
Can you point me towards docs on this file/plugins for x-pack in general?

Reply:
Some (hopefully) helpful links:
- https://github.com/elastic/kibana/blob/master/src/core/CONVENTIONS.md#plugin-structure - has some helpful notes about plugin structure/architecture 
- https://github.com/elastic/kibana/blob/master/src/core/PRINCIPLES.md
- Example plugin snippets from Kibana's docs: https://www.elastic.co/guide/en/kibana/master/migrating-legacy-plugins-examples.html#_creating_a_kibana_platform_plugin (in the context of migrating a legacy plugin to the new platform, which almost all plugins are on now, but still has some decent info to wade through)

LMK if you'd like me to explain more in person/more succinctly at the sync today!